### PR TITLE
Implement the 'unicode' inspection in librpminspect

### PR DESCRIPTION
**Pull request by dcantrell**
CVE-2021-42574 and CVE-2021-42694 led to the creation of the 'unicode' inspection in librpminspect.  This inspection checks %prep'ed source trees and all text files in source RPM packages for forbidden Unicode code points as defined in the rpminspect configuration file.

### CODE REVIEW

File: data/generic.yaml [hunk 1]
```diff
@@ -743,3 +744,43 @@ types:
...
+    # Regular expression to match file and directory names for
+    # exclusion.  Take care with making this too complicated as it may
+    # become difficult to maintain over time.  The regular expression
+    # here is processed by regex(3) and can use POSIX Extended Regular
+    # Expression syntax.
+    exclude: ^(\.git|\.hg|\.desktop|\.ppd|\.txt|\.directory|ChangeLog|NEWS)$
+
+    # List of file MIME types of files this inspection should exclude.
+    # MIME types are determined by libmagic, which is how the file(1)
+    # command works.  Sometimes matching by MIME type does not quit
```
Comment (jimbair) [inline data/generic.yaml:758]: nit: s/quit/quite/

File: include/constants.h [hunk 2]
```diff
@@ -427,12 +427,28 @@
...
-#define SPEC_TAG_RELEASE "Release:"
-#define SPEC_DISTTAG "%{?dist}"
+#define SPEC_SECTION_FILES     "%files"
+#define SPEC_TAG_RELEASE       "Release:"
+#define SPEC_DISTTAG           "%{?dist}"
+
+/*
+ * Names of rpmbuild(1) subdirectories.  rpminspect models an rpmbuild
+ * layout similar to how rpmbuild is used for distributions like
+ * Fedora Linux.  That is there is a top level directory (topdir) and
```
Comment (jimbair) [inline include/constants.h:440]: That is there is reads a little bumpy; I might reword this to something like

In this example, there is...
In Fedora, there is...

Something like that.

File: include/inspect.h [hunk 3]
```diff
@@ -508,6 +508,15 @@ bool inspect_badfuncs(struct rpminspect *ri);
  */
 bool inspect_runpath(struct rpminspect *ri);
 
+/**
+ * @brief Main driver for the 'unicode' inspection.
+ *
+ *
+ * @param ri Pointer to the struct rpminspect for the program.
```
Comment (jimbair) [inline include/inspect.h:515]: To the struct rpminspect uses? It feels like it's missing a word.

File: data/generic.yaml (hunk 1 above)
Comment (dcantrell) [reply data/generic.yaml]: Thanks, fixed.

File: include/constants.h (hunk 2 above)
Comment (dcantrell) [reply include/constants.h:440]: Oh yeah, that's a terrible comment.  Fix coming.

Comment (dcantrell) [reply include/constants.h:440]:
```
/*                                                                                                                                                            
 * Names of rpmbuild(1) subdirectories.  rpminspect models an rpmbuild                                                                                        
 * layout similar to how rpmbuild is used for distributions like                                                                                              
 * Fedora Linux.  The main idea is a top level directory is defined                                                                                           
 * (topdir) and all of the other directories used by rpmbuild are                                                                                             
 * within topdir.  For the purposes of rpminspect, the topdir                                                                                                 
 * definition here is a subdirectory in the working directory location                                                                                        
 * rpminspect is using.                                                                                                                                       
 */
```

Eh?

File: include/inspect.h (hunk 3 above)
Comment (dcantrell) [reply include/inspect.h:515]: This is a placeholder for the Doxygen work I plan to do for the library.  That's why it says '@param' and has a blank spot in the middle.  I just copy one from above and change it and always remind myself that I will come back and complete the Doxygen work.

In this context it's 'struct rpminspect' as in the data type.  Again, I will clean this up with the mega doxygen work coming soon.

**Comment by jimbair** [comment]
That's all I've got - looks good besides the small nits outlined above :)

**Pull request merged**